File: Ex3/Ex3.swift
import Foundation

// Функция для проверки, является ли число простым
func isPrime(num: Int) -> Bool {
    if num <= 1 {
        return false
    }
    for i in 2...Int(sqrt(Double(num))) {
        if num % i == 0 {
            return false
        }
    }
    return true
}

// Функция для решения задачи
func countPrimes(n: Int) -> Int {
    var count = 0
    for _ in 0..<n {
        print("Enter the number: ", terminator: "")
        if let input = readLine(), let num = Int(input), isPrime(num: num) {
            count += 1
        } else {
            continue
        }
    }
    print("Number of prime numbers: ", terminator: "")
    return count
}

func main() {
    print("Enter the number of numbers: ", terminator: "")
    if let input = readLine(), let n = Int(input) {
        print(countPrimes(n: n))
    } else {
        print("Invalid input. Please enter a number.")
    }
}

main()

File: Ex3/Ex3.py
# Функция для проверки, является ли число простым
def is_prime(num):
    if num <= 1:
        return False
    for i in range(2, int(num**0.5) + 1):
        if num % i == 0:
            return False
    return True

# Функция для решения задачи
def count_primes(n):
    count = 0
    for _ in range(n):
        num = int(input("Enter the number: "))
        if is_prime(num):
            count += 1
    print("Number of prime numbers: ", end="")
    return count

def main():
    n = int(input("Enter the number of numbers: "))
    print(count_primes(n))

# Запуск основной функции
main()